src/zkp/rangeproof.rs

```diff
@@ -207,6 +208,17 @@ impl ::core::fmt::Display for RangeProof {
     }
 }
 
+impl std::str::FromStr for RangeProof {
+    type Err = Error;
+    fn from_str(s: &str) -> Result<RangeProof, Error> {
+        let mut res = Vec::new();
```

Comment:
Could initialize this with `Vec::with_capacity(s.len() / 2)` to avoid repeated allocations as the string is decoded.